src/file/tokio_compat_file.rs

```diff
@@ -351,7 +351,11 @@ impl AsyncSeek for TokioCompatFile {
             if new_offset < prev_offset {
                 this.buffer.clear();
             } else if let Ok(offset) = (new_offset - prev_offset).try_into() {
-                self.consume(offset);
+                if offset > this.buffer.len() {
+                    this.buffer.clear();
+                } else {
+                    self.consume(offset);
+                }
```

Comment:
```suggestion
                if offset > this.buffer.len() {
                    this.buffer.clear();
                } else {
                    this.buffer.advance(offset);
                }
                this.inner.offset += offset.try_into().unwrap();
```

Reply:
We have increased the offset in https://github.com/openssh-rust/openssh-sftp-client/blob/ec5789501971ed6a5bd8382c2d02d185a21f9bb0/src/file/tokio_compat_file.rs#L343

This will increase the offset twice.

Reply:
Aha right.

But note that `<TokioCompatFile as AsyncBufRead::consume` also increases the offset.

I've updated my suggestion to not increase the offset.

Reply:
@silver-ymz I think we also need to fix `<File as AsyncBufRead>::consume` and `File as AsyncSeek`.